Can't install zookeeper on Mac OS X 10.9

/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.  -I./include -I./tests -I./generated  -Wall -Werror  -g -O2 -D_GNU_SOURCE -MT zookeeper.lo -MD -MP -MF .deps/zookeeper.Tpo -c -o zookeeper.lo `test -f 'src/zookeeper.c' || echo './'`src/zookeeper.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./include -I./tests -I./generated -Wall -Werror -g -O2 -D_GNU_SOURCE -MT zookeeper.lo -MD -MP -MF .deps/zookeeper.Tpo -c src/zookeeper.c  -fno-common -DPIC -o zookeeper.o
clang: error: argument unused during compilation: '-I .'
clang: error: argument unused during compilation: '-I ./include'
clang: error: argument unused during compilation: '-I ./tests'
clang: error: argument unused during compilation: '-I ./generated'
make[2]: **\* [zookeeper.lo] Error 1
make[1]: **\* [all] Error 2
Unable to build zookeeper library
make: **\* [.] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:810:12)
gyp ERR! System Darwin 13.3.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/foo/bar/lib/node_modules/zookeeper
gyp ERR! node -v v0.10.30
gyp ERR! node-gyp -v v0.13.1
gyp ERR! not ok
npm ERR! zookeeper@3.4.6-3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zookeeper@3.4.6-3 install script.
npm ERR! This is most likely a problem with the zookeeper package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls zookeeper
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.3.0
npm ERR! command "node" "/usr/local/bin/npm" "install"
npm ERR! cwd /Users/foo/bar
npm ERR! node -v v0.10.30
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/foo/bar/lib/npm-debug.log
npm ERR! not ok code 0

--- a/libzk-build.sh
+++ b/libzk-build.sh
@@ -44,7 +44,24 @@ if [ "$PLATFORM" != "SunOS" ]; then
 
     cd $BUILD_TMP
 
-    tar -zxf $ZK_FILE && \
+    tar -zxf $ZK_FILE
+
+    if [ "$PLATFORM" = "Darwin" ]; then
+        OS_X_VERSION=`sw_vers -productVersion`
+        if [ $OS_X_VERSION = "10.10" ]; then
+            PATCH_URL=https://issues.apache.org/jira/secure/attachment/12673212/ZOOKEEPER-2049.noprefix.trunk.patch
+            PATCH_FILE=/$BUILD_TMP/trunk.patch
+            echo "Downloading yosemite patch"
+            curl --silent --output $PATCH_FILE $PATCH_URL || wget $PATCH_URL -O $PATCH_FILE
+            if [ $? != 0 ] ; then
+                echo "Unable to download yosemite patch"
+                exit 1
+            fi
+            echo "Applying patch"
+            (cd $ZK && patch -p0 < $PATCH_FILE)
+        fi
+    fi
+
     cd $ZK/src/c && \
     ./configure \
         --without-syncapi \

